### PR TITLE
Update the hardcoded net3.1 to the current framework

### PR DESCRIPTION
**Pull request by v-wuzhai**

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibraryWithOSSupportedVersion.cs [hunk 1]
```diff
@@ -213,7 +213,7 @@ public void WhenTargetingWindowsSupportedOSPlatformVersionPropertyIsPreferredOve
         }
 
         [Theory]
-        [InlineData("netcoreapp3.1")]
+        [InlineData(ToolsetInfo.CurrentTargetFramework)]
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibraryWithOSSupportedVersion.cs:216]: This should not be updated.  The test is testing pre .NET 5 behavior for how target platforms were handled.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs [hunk 2]
```diff
@@ -720,7 +720,7 @@ public void It_includes_repository_url(bool privateRepo)
             var buildCommand = new BuildCommand(testAsset);
             buildCommand.Execute().Should().Pass();
 
-            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory("netcoreapp3.1").FullName, testProject.Name + ".dll");
+            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory(ToolsetInfo.CurrentTargetFramework).FullName, testProject.Name + ".dll");
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs:723]: I'd suggest this change:

```suggestion
            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory(testProject.TargetFrameworks).FullName, testProject.Name + ".dll");
```

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs [hunk 3]
```diff
@@ -326,7 +326,7 @@ static void Main(string[] args)
             {
                 Name = "ProjectC",
                 IsExe = true,
-                TargetFrameworks = "netstandard2.1;netcoreapp3.1"
+                TargetFrameworks = $"netstandard2.1;{ToolsetInfo.CurrentTargetFramework}"
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs:329]: Minor suggestions:

```suggestion
                TargetFrameworks = $"netstandard2.1;{tfm}"
```

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs [hunk 4]
```diff
@@ -337,7 +337,7 @@ static void Main(string[] args)
                 {
                     var ns = p.Root.Name.Namespace;
                     var itemGroup = new XElement(ns + "ItemGroup",
-                        new XAttribute("Condition", @"'$(TargetFramework)' == 'netcoreapp3.1'"));
+                        new XAttribute("Condition", $@"'$(TargetFramework)' == '{ToolsetInfo.CurrentTargetFramework}'"));
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs:340]:
```suggestion
                        new XAttribute("Condition", $@"'$(TargetFramework)' == '{tfm}'"));
```

**Review by dsplaisted — DISMISSED** [review]
Most changes look good, but there's one test that should not be updated.

**Review by marcpopMSFT — APPROVED** [review]

**Pull request merged**